### PR TITLE
[JSDoc] Add missing documentation for botbuilder-core files

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses # 2602

## Description
This PR adds the missing documentation in [botbuilder-core](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder-core) library.
Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

## Specific Changes
- Updated jsdoc/require-jsdoc in .eslintrc.json from warn to error.
- Added JSDoc comments in all public methods and classes.

## Testing
The following image shows the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

**Review by sw-joelmut — APPROVED** [review]

File: libraries/botbuilder-core/src/cardFactory.ts [hunk 1]
```diff
@@ -156,6 +156,15 @@ export class CardFactory {
         buttons?: (CardAction | string)[],
         other?: Partial<HeroCard>
     ): Attachment;
+    /**
+     * Returns an attachment for a hero card.
+     *
```
Comment (santgr11) [inline libraries/botbuilder-core/src/cardFactory.ts:161]: This space between the summary and the params is not present in the rest of the comments

File: libraries/botbuilder-core/src/cardFactory.ts [hunk 2]
```diff
@@ -276,6 +285,15 @@ export class CardFactory {
         buttons?: (CardAction | string)[],
         other?: Partial<ThumbnailCard>
     ): Attachment;
+    /**
+     * Returns an attachment for a thumbnail card.
+     *
```
Comment (santgr11) [inline libraries/botbuilder-core/src/cardFactory.ts:290]: same here

**Review by iscaltritti-sw — APPROVED** [review]

File: libraries/botbuilder-core/src/cardFactory.ts (hunk 1 above)
Comment (ceciliaavila) [reply libraries/botbuilder-core/src/cardFactory.ts:161]: It's like this in the previous overloads of this method, that's why I kept it.

File: libraries/botbuilder-core/src/cardFactory.ts (hunk 2 above)
Comment (ceciliaavila) [reply libraries/botbuilder-core/src/cardFactory.ts:290]: It's like this in the previous overloads of this method, that's why I kept it.

**Review by santgr11 — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**